addons/point_of_sale/static/src/app/screens/product_screen/product_screen.js

```diff
@@ -385,6 +385,15 @@ export class ProductScreen extends Component {
         return Array.from(new Set(products));
     }
 
+    getProductCartQty(product) {
+        const productTmplId = product.id;
+
+        return this.currentOrder.lines.reduce(
+            (acc, ol) => (ol.product_id.product_tmpl_id.id === productTmplId ? acc + ol.qty : acc),
+            0
+        );
+    }
+
```

Comment:
In a performance pov you can maybe do something like that to avoid looping on each orderline for each product

```js
onWillRender(() => {
    // If its a shared order it can be paid from another POS
    if (this.currentOrder?.state !== "draft") {
        this.pos.add_new_order();
    }

    this.state.quantityByProductTmplId = this.currentOrder.lines.reduce((acc, ol) => {
        acc[ol.product_id.product_tmpl_id.id]
            ? (acc[ol.product_id.product_tmpl_id.id] += ol.qty)
            : (acc[ol.product_id.product_tmpl_id.id] = ol.qty);
        return acc;
    }, {});
});
```

Reply:
@davidmonnom ,
Changes are done.
Could please check?